app/lib/core/services/database/firebase_database_interface.dart

```diff
@@ -10,10 +11,29 @@ import 'database_interface.dart';
 // TODO: #130. Replace MySQL and SQLFLite with Firebase Database calls.
 // We also probably need a separate service for FireBaseAuthentication
 // Up to you if these are in the same file
-class FirebaseDatabaseInterface extends DatabaseInterface {
+class FirebaseDatabaseInterface implements DatabaseInterface {
+  final FirebaseFirestore _firestore;
```

Comment:
```suggestion
  // We don't use locator for Firestore because mocking requires the instance to be under out control
  // So, we take in the instance in the constructor for testability
  final FirebaseFirestore _firestore;
```